<img src=".\images\Logo.png" width=150 align="left" /> <img src=".\images\Logo2.jpg" width=450 align="right" />


# <center><font color= #1e8449 > <b>CU34_Predicción de demanda de servicios</font></center>

<font color='steelblue'><b>Citizenlab Data Science Methodology > II - Data Processing Domain </font>
***
> # <font color='steelblue'> <b>05.- Data Collection</font>

<font color='steelblue'>Data Collection is the process to obtain and generate (if required) necessary data to model the problem.</font>

### 05. Unir todos los datos en un data.frame con repeticiones

* Unión de todas las tablas donde se reptirán los datos que no tengan la máxima resolución







<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Settings" data-toc-modified-id="Settings-1"><font color="green">Settings</font></a></span></li><li><span><a href="#Data-Load" data-toc-modified-id="Data-Load-2"><font color="green">Data Load</font></a></span></li><li><span><a href="#ETL-Processes" data-toc-modified-id="ETL-Processes-3">ETL Processes</a></span><ul class="toc-item"><li><span><a href="#Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files" data-toc-modified-id="Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files-3.1">Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files</a></span></li></ul></li><li><span><a href="#Synthetic-Data-Generation" data-toc-modified-id="Synthetic-Data-Generation-4">Synthetic Data Generation</a></span></li><li><span><a href="#Fake-Data-Generation" data-toc-modified-id="Fake-Data-Generation-5">Fake Data Generation</a></span></li><li><span><a href="#Open-Data" data-toc-modified-id="Open-Data-6">Open Data</a></span></li><li><span><a href="#Data-Save" data-toc-modified-id="Data-Save-7"><font color="green">Data Save</font></a></span></li><li><span><a href="#Main-Conclusions" data-toc-modified-id="Main-Conclusions-8"><font color=" #2874a6 ">Main Conclusions</font></a></span></li><li><span><a href="#Main-Actions" data-toc-modified-id="Main-Actions-9"><font color=" #2874a6 ">Main Actions</font></a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Acciones-done" data-toc-modified-id="Acciones-done-9.0.1"><font color=" #2874a6 ">Acciones done</font></a></span></li><li><span><a href="#Acctions-to-perform" data-toc-modified-id="Acctions-to-perform-9.0.2"><font color=" #2874a6 ">Acctions to perform</font></a></span></li></ul></li></ul></li></ul></div>

## <font color='green'>Settings</font>

### Encoding

Con la siguiente expresión se evitan problemas con el encoding al ejecutar el notebook. Es posible que deba ser eliminada o adaptada a la máquina en la que se ejecute el código.

In [32]:
Sys.setlocale(category = "LC_ALL", locale = "es_ES.UTF-8")

[1] "es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/C"

### Packages to use


- {tcltk} para selección interactiva de archivos locales
- {readr} para leer y escribir archivos csv
- {dplyr} para explorar datos


In [33]:
library(readr)
library(dplyr)




### Paths

In [34]:
iPath <- "Data/Input/"
oPath <- "Data/Output/"

## <font color='green'>Data Load</font>

If there are more than one input file, make as many sections as files to import.

Instrucciones
- Los ficheros de entrada del proceso están siempre en Data/Input/.   
- Si hay más de un fichero de entrada, se crean tantos objetos iFile_xx y file_data_xx como ficheros de entrada (xx número correlativo con dos dígitos, rellenar con ceros a la izquierda)

1. Datos por día, sección y servicio

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [35]:
# file_data_01 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [36]:
iFile_01 <- "CU_34_05_02_02_demanda_sim.csv"
file_data_01 <- paste0(iPath, iFile_01)

if(file.exists(file_data_01)){
    cat("Se leerán datos del archivo: ", file_data_01)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_02_02_demanda_sim.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [37]:
data_01 <- read_csv(file_data_01)


Rows: 274784 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (4): Servicio, CMUN, CDIS, CSEC
dbl  (2): Futbol, nservicios
date (1): Fecha

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [38]:
data_01 |> glimpse()

Rows: 274,784
Columns: 7
$ Fecha      <date> 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01…
$ Servicio   <chr> "Taxi", "Taxi", "Taxi", "Taxi", "Taxi", "Taxi", "Taxi", "Ta…
$ CMUN       <chr> "001", "002", "002", "003", "004", "004", "004", "004", "00…
$ CDIS       <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01", "01",…
$ CSEC       <chr> "001", "001", "002", "001", "001", "002", "003", "004", "00…
$ Futbol     <dbl> 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,…
$ nservicios <dbl> 0, 0, 0, 50, 0, 50, 50, 0, 2, 0, 0, 7, 0, 1, 4, 7, 10, 50, …


Muestra de datos:

In [39]:
data_01 |> slice_head(n = 5)

Fecha      Servicio CMUN CDIS CSEC Futbol nservicios
1 2022-01-01 Taxi     001  01   001  0       0        
2 2022-01-01 Taxi     002  01   001  0       0        
3 2022-01-01 Taxi     002  01   002  0       0        
4 2022-01-01 Taxi     003  01   001  1      50        
5 2022-01-01 Taxi     004  01   001  0       0

2. Datos por municipio y servicio

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [40]:
# file_data <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [41]:
iFile_02 <- "CU_34_05_02_03_capacidad_sim.csv"
file_data_02 <- paste0(iPath, iFile_02)

if(file.exists(file_data_02)){
    cat("Se leerán datos del archivo: ", file_data_02)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_02_03_capacidad_sim.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [42]:
data_02 <- read_csv(file_data_02)


Rows: 358 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): CMUN, Servicio
dbl (1): capacidad

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [43]:
data_02 |> glimpse()

Rows: 358
Columns: 3
$ CMUN      <chr> "001", "001", "002", "002", "003", "003", "004", "004", "005…
$ Servicio  <chr> "Taxi", "Delivery", "Taxi", "Delivery", "Taxi", "Delivery", …
$ capacidad <dbl> 50, 50, 50, 59, 50, 61, 50, 56, 65, 70, 66, 75, 64, 70, 50, …


Muestra de datos:

In [44]:
data_02 |> slice_head(n = 5)

CMUN Servicio capacidad
1 001  Taxi     50       
2 001  Delivery 50       
3 002  Taxi     50       
4 002  Delivery 59       
5 003  Taxi     50

3. Datos por fecha y sección

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [45]:
# file_data_03 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [46]:
iFile_03 <- "CU_34_05_04_meteo_secciones.csv"
file_data_03 <- paste0(iPath, iFile_03)

if(file.exists(file_data_03)){
    cat("Se leerán datos del archivo: ", file_data_03)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_04_meteo_secciones.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [47]:
data_03 <- read_csv(file_data_03)


Rows: 137392 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (3): CMUN, CDIS, CSEC
dbl  (4): tmed, prec, velmedia, presMax
date (1): fecha

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [48]:
data_03 |> glimpse()

Rows: 137,392
Columns: 8
$ CMUN     <chr> "001", "002", "002", "003", "004", "004", "004", "004", "005"…
$ CDIS     <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01", "01", "…
$ CSEC     <chr> "001", "001", "002", "001", "001", "002", "003", "004", "001"…
$ fecha    <date> 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, …
$ tmed     <dbl> 9.472965, 8.170448, 8.526402, 10.012282, 10.574368, 10.508450…
$ prec     <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…
$ velmedia <dbl> 0.8229741, 0.9750191, 0.8874318, 2.3162929, 0.6970652, 0.6875…
$ presMax  <dbl> 901.6717, 970.1249, 967.2948, 849.5611, 964.2144, 965.4248, 9…


Muestra de datos:

In [49]:
data_03 |> slice_head(n = 5)

CMUN CDIS CSEC fecha      tmed      prec velmedia  presMax 
1 001  01   001  2022-01-01  9.472965 0    0.8229741 901.6717
2 002  01   001  2022-01-01  8.170448 0    0.9750191 970.1249
3 002  01   002  2022-01-01  8.526402 0    0.8874318 967.2948
4 003  01   001  2022-01-01 10.012282 0    2.3162929 849.5611
5 004  01   001  2022-01-01 10.574368 0    0.6970652 964.2144

4. Datos por sección

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [50]:
# file_data_04 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [51]:
iFile_04 <- "CU_34_05_02_01_indicadores_secc_extendido.csv"
file_data_04 <- paste0(iPath, iFile_04)

if(file.exists(file_data_04)){
    cat("Se leerán datos del archivo: ", file_data_04)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_02_01_indicadores_secc_extendido.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [52]:
data_04 <- read_csv(file_data_04)


Rows: 4440 Columns: 24
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (4): CMUN, dist, secc, NSEC
dbl (20): ccaa, CPRO, t1_1, t2_1, t2_2, t3_1, t4_1, t4_2, t4_3, t5_1, t6_1, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [53]:
data_04 |> glimpse()

Rows: 4,440
Columns: 24
$ ccaa             <dbl> 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 1…
$ CPRO             <dbl> 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 2…
$ CMUN             <chr> "001", "002", "002", "003", "004", "004", "004", "004…
$ dist             <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01",…
$ secc             <chr> "001", "001", "002", "001", "001", "002", "003", "004…
$ t1_1             <dbl> 55, 2358, 2435, 248, 2886, 3376, 2161, 1523, 1648, 10…
$ t2_1             <dbl> NA, 0.4724, 0.4830, 0.3952, 0.5135, 0.4793, 0.5016, 0…
$ t2_2             <dbl> NA, 0.5276, 0.5170, 0.6048, 0.4865, 0.5207, 0.4984, 0…
$ t3_1             <dbl> NA, 40.5161, 38.3339, 47.4597, 44.4099, 40.6810, 38.0…
$ t4_1             <dbl> NA, 0.1425, 0.1959, 0.1371, 0.1611, 0.1739, 0.2230, 0…
$ t4_2             <dbl> NA, 0.7443, 0.7002, 0.6573, 0.6067, 0.6727, 0.6395, 0…
$ t4_3             <dbl> NA, 0.1132, 0.1039, 0.2056, 0.2322, 0.1534, 0.1374, 0…
$ t5_1          

Muestra de datos:

In [54]:
data_04 |> slice_head(n = 5)

ccaa CPRO CMUN dist secc t1_1 t2_1   t2_2   t3_1    t4_1   ... t7_1   t8_1  
1 13   28   001  01   001    55     NA     NA      NA     NA ...     NA     NA
2 13   28   002  01   001  2358 0.4724 0.5276 40.5161 0.1425 ... 0.4016 0.3971
3 13   28   002  01   002  2435 0.4830 0.5170 38.3339 0.1959 ... 0.4244 0.4224
4 13   28   003  01   001   248 0.3952 0.6048 47.4597 0.1371 ... 0.1542 0.1449
5 13   28   004  01   001  2886 0.5135 0.4865 44.4099 0.1611 ... 0.1470 0.1409
  t9_1   t10_1  t11_1  t12_1  NSEC                       area     elevation
1     NA     NA     NA     NA Acebeda, La - 01.001       21848791 1401     
2 0.4377 0.0912 0.6063 0.6672 Ajalvir - 01.001           15585050  653     
3 0.4438 0.1226 0.5955 0.6788 Ajalvir - 01.002            4148273  715     
4 0.5280 0.0932 0.5000 0.5514 Alameda del Valle - 01.001 25674491 1150     
5 0.2602 0.1814 0.4213 0.5147 Álamo, El - 01.001          5046680  593     
  densidad_hab_km2
1   2.517302      
2 151.298837      
3 586.991204      
4   9.659393      
5 571.861060

## ETL Processes

### Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files

Se han importado en el apartado Data Load anterior:

* Datos por fecha, servicio y sección (demanda)
* Datos por munipio (capacidad)
* Datos por fecha y sección (meteorológicos)
* Datos por sección (indicadores)

Incluir apartados si procede para: Extracción de datos (select, filter), Transformación de datos, (mutate, joins, ...). Si es necesario tratar datos perdidos, indicarlo también en NB 09.2

Si no aplica: Estos datos no requieren tareas de este tipo.

#### Data transformation

In [55]:
colnames(data_01)
colnames(data_02)
colnames(data_03)
colnames(data_04)

[1] "Fecha"      "Servicio"   "CMUN"       "CDIS"       "CSEC"      
[6] "Futbol"     "nservicios"

[1] "CMUN"      "Servicio"  "capacidad"

[1] "CMUN"     "CDIS"     "CSEC"     "fecha"    "tmed"     "prec"     "velmedia"
[8] "presMax"

[1] "ccaa"             "CPRO"             "CMUN"             "dist"            
 [5] "secc"             "t1_1"             "t2_1"             "t2_2"            
 [9] "t3_1"             "t4_1"             "t4_2"             "t4_3"            
[13] "t5_1"             "t6_1"             "t7_1"             "t8_1"            
[17] "t9_1"             "t10_1"            "t11_1"            "t12_1"           
[21] "NSEC"             "area"             "elevation"        "densidad_hab_km2"

In [56]:
data <- data_01 |> 
    full_join(data_02, by = c("CMUN", "Servicio")) |>
    full_join(data_03, by = c("Fecha" = "fecha", "CMUN", "CDIS", "CSEC")) |>
    full_join(data_04, by = c("CMUN", "CDIS" = "dist", "CSEC" = "secc"))

In [57]:
data |> glimpse()

Rows: 274,792
Columns: 33
$ Fecha            <date> 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, 2022…
$ Servicio         <chr> "Taxi", "Taxi", "Taxi", "Taxi", "Taxi", "Taxi", "Taxi…
$ CMUN             <chr> "001", "002", "002", "003", "004", "004", "004", "004…
$ CDIS             <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01",…
$ CSEC             <chr> "001", "001", "002", "001", "001", "002", "003", "004…
$ Futbol           <dbl> 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,…
$ nservicios       <dbl> 0, 0, 0, 50, 0, 50, 50, 0, 2, 0, 0, 7, 0, 1, 4, 7, 10…
$ capacidad        <dbl> 50, 50, 50, 50, 50, 50, 50, 50, 65, 65, 65, 65, 65, 6…
$ tmed             <dbl> 9.472965, 8.170448, 8.526402, 10.012282, 10.574368, 1…
$ prec             <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ velmedia         <dbl> 0.8229741, 0.9750191, 0.8874318, 2.3162929, 0.6970652…
$ presMax          <dbl> 901.6717, 970.1249, 967.2948, 849.5611, 964.2144, 965…
$ ccaa        

In [58]:
data |> slice_head(n = 5)

Fecha      Servicio CMUN CDIS CSEC Futbol nservicios capacidad tmed      prec
1 2022-01-01 Taxi     001  01   001  0       0         50         9.472965 0   
2 2022-01-01 Taxi     002  01   001  0       0         50         8.170448 0   
3 2022-01-01 Taxi     002  01   002  0       0         50         8.526402 0   
4 2022-01-01 Taxi     003  01   001  1      50         50        10.012282 0   
5 2022-01-01 Taxi     004  01   001  0       0         50        10.574368 0   
  ... t7_1   t8_1   t9_1   t10_1  t11_1  t12_1  NSEC                      
1 ...     NA     NA     NA     NA     NA     NA Acebeda, La - 01.001      
2 ... 0.4016 0.3971 0.4377 0.0912 0.6063 0.6672 Ajalvir - 01.001          
3 ... 0.4244 0.4224 0.4438 0.1226 0.5955 0.6788 Ajalvir - 01.002          
4 ... 0.1542 0.1449 0.5280 0.0932 0.5000 0.5514 Alameda del Valle - 01.001
5 ... 0.1470 0.1409 0.2602 0.1814 0.4213 0.5147 Álamo, El - 01.001        
  area     elevation densidad_hab_km2
1 21848791 1401        2.517302      
2 15585050  653      151.298837      
3  4148273  715      586.991204      
4 25674491 1150        9.659393      
5  5046680  593      571.861060

## Synthetic Data Generation

No aplica

## Fake Data Generation

No aplica

## Open Data

No aplica

## <font color='green'>Data Save</font>

Este proceso, puede copiarse y repetirse en aquellas partes del notebbok que necesiten guardar datos.
Recuerde cambiar las cadenas añadida del fichero para diferenciarlas

<font color='tomato'> Identificamos los datos a guardar</font>

In [59]:
data_to_save <- data


<font color='tomato'>Estructura de nombre de archivos:</font>

* Código del caso de uso, por ejemplo "CU_04"
* Número del proceso que lo genera, por ejemplo "_05".
* Número de la tarea que lo genera, por ejemplo "_01"
* En caso de generarse varios ficheros en la misma tarea, llevarán _01 _02 ... después
* Nombre: identificativo de "properData", por ejemplo "_zonasgeo"
* Extensión del archivo

Ejemplo: "CU_04_05_01_01_zonasgeo.json, primer fichero que se genera en la tarea 01 del proceso 05 (Data Collection) para el caso de uso 04 (vacunas)

Importante mantener los guiones bajos antes de proceso, tarea, archivo y nombre

### Proceso 05

In [60]:
caso <- "CU_34"
proceso <- '_05'
tarea <- "_05"
archivo <- ""
proper <- "_servicios_completo"
extension <- ".csv"

<font color='tomato'><b> OPCION A:</b> Uso del paquete "tcltk" para mayor comodidad</font>

* Buscar carpeta, escribir nombre de archivo SIN extensión (se especifica en el código)
* Especificar sufijo2 si es necesario
* Cambiar datos por datos_xx si es necesario

In [ ]:
# file_save <- paste0(caso, proceso, tarea, tcltk::tkgetSaveFile(), proper, extension) 
# path_out <- paste0(oPath, file_save)
# write_csv(data_to_save, path_out)

# cat('File saved as: ')
# path_out

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

- Los ficheros de salida del proceso van siempre a Data/Output/.  

In [61]:
file_save <- paste0(caso, proceso, tarea, archivo, proper, extension) 
path_out <- paste0(oPath, file_save)
write_csv(data_to_save, path_out)

cat('File saved as: ')
path_out

File saved as: 

[1] "Data/Output/CU_34_05_05_servicios_completo.csv"

#### Copia del fichero a Input

Si el archivo se va a usar en otros notebooks, copiar a la carpeta Input

In [62]:
path_in <- paste0(iPath, file_save)
file.copy(path_out, path_in, overwrite = TRUE)

[1] TRUE

## <font color=' #2874a6 '>Main Conclusions</font>

<font color='steelblue'>List and describe the general conclusions of the analysis carried out​.</font>

### Prerequisites

Para que funcione este código se necesita:

* Las rutas de archivos `Data/Input` y `Data/Output` deben existir (relativas a la ruta del _notebook_)
* El paquete tcltk instalado para seleccionar archivos interactivamente. No se necesita en producción.
* Los paquetes readr, dplyr deben estar instalados.


### Configuration Management

This notebook has been tested with the following versions of R and packages. It cannot be assured that later versions work in the same way:
* R 4.2.2
* tcltk 4.2.2
* readr 2.1.3
* dplyr 1.0.10

### Data structures

#### Objeto `data`

- Hay 274792 filas con información de las siguientes variables:
	* Fecha
	* Servicio
	* CMUN
	* CDIS
	* CSEC
	* Futbol
	* nservicios
	* capacidad
	* tmed
	* prec
	* velmedia
	* presMax
	* ccaa
	* CPRO
	* t1_1
	* t2_1
	* t2_2
	* t3_1
	* t4_1
	* t4_2
	* t4_3
	* t5_1
	* t6_1
	* t7_1
	* t8_1
	* t9_1
	* t10_1
	* t11_1
	* t12_1
	* NSEC
	* area
	* elevation
	* densidad_hab_km2


#### Observaciones generales sobre los datos

- Los datos se repiten en aquellas columnas que no tienen la misma resolución que el conjunto más amplio, es decir:
    - Los datos de capacidad son los mismos para todos los municipios todos los días
    - Los datos de indicadores son los mismos para todos los días en la misma sección
- El fichero incluye algunos datos reales y otros sintéticos


### Consideraciones para despliegue en piloto

* Este es el fichero principal que contiene todos los datos del caso
* Si llegaran datos reales solicitados a la empresa, habría que sustituir los sintéticos por los reales
* Para la representación en mapas, se deben combinar los datos del csv con los datos del json en la implementación que se haga

### Consideraciones para despliegue en producción

* Se deben crear los procesos ETL en producción necesarios para que los datos de entrada estén actualizados
* Se deben conseguir datos reales para sustituir los sintéticos
* Para la representación en mapas, se deben combinar los datos del csv con los datos del json de la primera trea en la implementación que se haga

## <font color=' #2874a6 '>Main Actions</font>

#### <font color=' #2874a6 '>Acciones done</font>

<font color='steelblue'>Indicate the actions that have been carried out in this process</font>

- Se han unido las tablas de datos del caso de uso en un único archivo

#### <font color=' #2874a6 '>Acctions to perform</font>

<font color='steelblue'>Indicate the actions that must be carried out in subsequent processes</font>

- Se deben realizar el resto de procesos de la dimensión Data Processing

## <font color='RED'>CODE TO DEPLOY (PILOT)</font>

A continuación se incluirá el código que deba ser llevado a despliegue para producción, dado que se entiende efectúa operaciones necesarias sobre los datos en la ejecución del prototipo

<font color=' red '><b>Description</b></font>

- No hay código que desplegar en el piloto, ya que estos datos son estáticos o en todo caso cambian con muy poca frecuencia, altamente improbable durante el proyecto.
- El fichero deberá estar incluido en la ruta de datos de la implementación


<font color=' red '><b>CODE</b></font>

In [ ]:
# incluir código